src/Client/Adapter/Socket.php

```diff
@@ -240,12 +240,19 @@ public function connect($host, $port = 80, $secure = false)
                 $flags |= STREAM_CLIENT_PERSISTENT;
             }
 
+            if (isset($this->config['connecttimeout'])) {
+                $connectTimeout = $this->config['connecttimeout'];
+            } elseif (isset($this->config['timeout'])) {
+                $connectTimeout = $this->config['timeout'];
+            } else {
+                $connectTimeout = @ini_get('default_socket_timeout');
```

Comment:
We generally don't use the `@` operator. Anyway, in this case the value `default_socket_timeout` is available since PHP 4.3.0.